### PR TITLE
Created Module-Related Tables

### PR DESCRIPTION
**Pull request by fru1tstand**
Module portion of the database schema looks like this:
http://i.imgur.com/IVoUkFA.png

Module Creation User flow:
1. User creates a module
2. User adds a lesson to the module
3. User adds one or more cards to the lesson
4. Repeat 2-3 until module is complete

Module Consuming User Flow:
1. User selects a module
2. User is introduced to/starts on a lesson (preferably in the order intended)
3. User interacts with cards, which send "oncomplete" requests to the server, dependent on the logic contained within the card.
4. Repeat 2-3 until module is complete

### CODE REVIEW

File: src/db/table/module.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  FOREIGN KEY (`language_id`)
+  REFERENCES `language` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE,
+
+  CONSTRAINT `fk_module_user_owner`
+  FOREIGN KEY (`owner_user_id`)
+  REFERENCES `user` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
```
Comment (jonathanpli) [inline src/db/table/module.sql:31]: If a contributor deletes their account, their modules shouldn't be deleted, right?

**Review by jonathanpli — CHANGES_REQUESTED** [review]
Looks good

File: src/db/table/module.sql (hunk 1 above)
Comment (fru1tstand) [reply src/db/table/module.sql:31]: I think that's a business decision rule. If a user deletes their account, they supposedly want all of their data removed. This includes modules they're owner to. Thoughts?

**Comment by fru1tstand** [comment]
IP Review

File: src/db/table/module.sql (hunk 1 above)
Comment (jonathanpli) [reply src/db/table/module.sql:31]: If other users are currently accessing the module though, they shouldn't lose all their progress, right? I think the owner can be removed (perhaps have an ExceptionNull general admin that assumes ownership in that case?) without it affecting the users interacting with the module

Comment (fru1tstand) [reply src/db/table/module.sql:31]: Might be a moral issue though. We could technically include something in our ToS that states "anything you post is now ours" or "deleting your account won't delete your modules", but that might make users a bit apprehensive about our platform. I believe in letting the user have 100% control of their content, and if that means deleting a module someone else is currently working on, so be it. We should loop in the rest of the team about this discussion though- @chengsuchen @gabealex.

Comment (jonathanpli) [reply src/db/table/module.sql:31]: Sounds good. We'll revisit

**Review by jonathanpli — APPROVED** [review]

**Comment by fru1tstand** [comment]
Opened #23.

**Pull request merged**